### DISCUSSION TITLE
Override cmp keybindings

### DISCUSSION THREAD

**Discussion by ivanuricardo** [NvChad/NvChad#2369] [Q&A] [answered]
Hello,
I am trying to override the tab mapping in cmp. More specifically, I want to replace <tab> and <S-tab> with <Down> and <Up>.  I looked at [this issue](https://github.com/NvChad/NvChad/issues/1236#issue-1275837914), but my lua doesn't run with the ["hrsh7th/nvim-cmp"] = function() syntax. Similarly, I looked at [this issue](https://github.com/NvChad/NvChad/issues/1393), but I can not figure out the tab mappings. Thanks!

**Comment by siduck** [comment, marked as answer]
@ivanuricardo just read the default cmp config in nvchad properly and then just index the <TAB> etc keys in the opts function of cmp  ( in  your custom plugins )

something like this 


```lua
  {
    "hrsh7th/nvim-cmp",
    opts = function()
      local conf = require "plugins.configs.cmp"

      -- your stuff
      conf.mapping["<TAB>"] = require("cmp").mapping.select_prev_item()
 
      return conf
    end,
  }, 
  ```
